ConnectorCore/DllConflictManagement/DllConflictManager.cs

```diff
@@ -101,19 +101,37 @@ HashSet<string> visitedAssemblies
     }
   }
 
-  private bool ShouldSkipCheckingConflictBecauseOfAssemblyLocation(Assembly loadedAssembly)
+  private static bool PathIsValid(string path)
   {
+    if (path.Any(x => Path.GetInvalidPathChars().Contains(x)))
+    {
+      return false;
+    }
+    return true;
+  }
+
+  private bool ShouldSkipCheckingConflictBecauseOfAssemblyLocation(Assembly? loadedAssembly)
+  {
+    if (string.IsNullOrWhiteSpace(loadedAssembly?.Location))
+    {
+      return false;
+    }
+    string location = loadedAssembly!.Location;
+    if (!PathIsValid(location))
+    {
+      return false;
+    }
```

Comment:
Do we want to log either of these exits?

Reply:
I don't want to log anything to our servers that doesn't matter.  We're logging too much as it is.

Reply:
If there's an unhandled exception it should surface - we're not expecting one I guess

Reply:
Might we want to check for NotSupportedException? https://learn.microsoft.com/en-us/dotnet/api/system.reflection.assembly.location?view=net-8.0

Reply:
maybe but it's likely already handled somewhere and it's not the current problem this is solving